### PR TITLE
fix(query): useridentity hostname only support char %

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

1. username containing @ should not encode when record log or err msg

2. We disabled arbitrary hostnames when creating users last year pr https://github.com/datafuselabs/databend/pull/11818 so the user meta key can only be generated by username, and % is a default value in UserIdentity.

3. delete StageFileInfo::creator

Note:

Also, do not support localhost as hostname.

- Fixes #[Link the issue here]

## Tests

- [ ] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [x] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/15652)

### CODE REVIEW

File: src/meta/app/src/principal/tenant_user_ident.rs [hunk 1]
```diff
@@ -70,7 +70,11 @@ mod tests {
 
     fn test_format_parse(user: &str, host: &str, expect: &str) {
         let tenant = Tenant::new_literal("test_tenant");
-        let user_ident = UserIdentity::new(user, host);
+        let user_ident = UserIdentity {
```
Comment (TCeason) [inline src/meta/app/src/principal/tenant_user_ident.rs:73]: test_tenant_user_ident_as_kvapi_key use this function. So use struct init replace new function.

File: src/meta/proto-conv/tests/it/user_proto_conv.rs [hunk 2]
```diff
@@ -806,7 +809,10 @@ fn test_old_stage_file() -> anyhow::Result<()> {
             NaiveDate::from_ymd_opt(2022, 9, 16).unwrap(),
             NaiveTime::from_hms_opt(0, 1, 2).unwrap(),
         );
-        let user_id = mt::principal::UserIdentity::new("datafuselabs", "datafuselabs.rs");
```
Comment (TCeason) [inline src/meta/proto-conv/tests/it/user_proto_conv.rs:809]: In order not to change the test result, hostname is left here. But do we need to construct useridentity here with a new method and change the test result at the same time cc @youngsofun @drmingdrmer

File: src/meta/app/src/principal/tenant_user_ident.rs (hunk 1 above)
Comment (TCeason) [reply src/meta/app/src/principal/tenant_user_ident.rs:73]: cc @drmingdrmer

**Review by drmingdrmer — APPROVED** [review]
> Reviewed 24 of 24 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/datafuselabs/databend/15652)* status: all files reviewed, 2 unresolved discussions

**Review by b41sh — APPROVED** [review]

**Comment by TCeason** [comment]
Some Old version use specifica host

**Pull request closed without merging**